server/py/framework/utils/clients/iguazio/v4.py

```
@@ -105,6 +108,49 @@ def refresh_access_tokens(
         # TODO: Implement this method once it is available in the Iguazio package
         pass
 
+    def revoke_offline_token(self, token: str) -> None:
+        """
+        Revoke an offline token in Iguazio.
+
+        This method sends a revoke request to Iguazio in order to invalidate
+        the provided offline token. Once revoked, the token can no longer be
+        used to obtain access tokens.
+
+        :param token: The offline token string to revoke.
+        :raises mlrun.errors.MLRunInvalidArgumentError: If the provided token is empty.
+        :raises mlrun.errors.MLRunUnauthorizedError: If the revocation request fails.
+        """
+        if not token:
+            raise mlrun.errors.MLRunInvalidArgumentError("Offline token is empty")
+
+        self._logger.info("Revoking offline token via Iguazio")
+
+        try:
+            # Use Iguazio client to revoke the token
+            options = RevokeOfflineTokenOptionsV1(token=token)
+            self._client.revoke_offline_token(options=options)
+            self._logger.info("Successfully revoked offline token via Iguazio")
+        except httpx.HTTPStatusError as exc:
+            error_message, ctx = self._extract_response_error(exc.response)
+            self._logger.warning(
+                "Failed to revoke offline token from Iguazio",
+                status_code=exc.response.status_code,
+                error_message=error_message,
+                ctx=ctx,
+                exc=mlrun.errors.err_to_str(exc),
+            )
+            raise mlrun.errors.MLRunUnauthorizedError(
```

How about adding `raise_on_error` option here or do we want to raise always?

---

I think we need to raise an error here anyway, because if the token cannot be deleted in Orca, it might indicate another problem, and this is our way to catch it